src/main/java/moze_intel/projecte/network/commands/KnowledgeCMD.java

```diff
@@ -0,0 +1,122 @@
+			)
+			.then(Commands.literal("test")
+				.requires(PEPermissions.COMMAND_KNOWLEDGE_TEST)
+				.then(executeWithParameters(ActionType.TEST, context))
+			);
+	}
+
+	private static ArgumentBuilder<CommandSourceStack, ?> executeWithParameters(ActionType actionType, CommandBuildContext context) {
+		return Commands.argument("player", EntityArgument.player())
+			.then(Commands.argument("item", ItemArgument.item(context))
+				.executes(ctx -> handle(ctx, actionType))
+			);
+	}
+
+	private static int handle(CommandContext<CommandSourceStack> ctx, ActionType action) throws CommandSyntaxException {
+		ServerPlayer player = EntityArgument.getPlayer(ctx, "player");
+		Optional<IKnowledgeProvider> cap = player.getCapability(PECapabilities.KNOWLEDGE_CAPABILITY).resolve();
+		if (cap.isEmpty()) {
+			ctx.getSource().sendFailure(PELang.COMMAND_PROVIDER_FAIL.translateColored(ChatFormatting.RED));
+			return 0;
+		}
+		IKnowledgeProvider provider = cap.get();
+		if (action == ActionType.CLEAR) {
+			if (provider.getKnowledge().size() == 0) {
+				ctx.getSource().sendFailure(PELang.COMMAND_KNOWLEDGE_CLEAR_FAIL.translateColored(ChatFormatting.RED, player.getDisplayName()));
+				return 0;
+			}
+			provider.clearKnowledge();
+			provider.sync(player);
+			ctx.getSource().sendSuccess(PELang.COMMAND_KNOWLEDGE_CLEAR_SUCCESS.translateColored(ChatFormatting.GREEN, player.getDisplayName()), true);
+			return 1;
+		}
+
+		ItemStack item = new ItemStack(ItemArgument.getItem(ctx, "item").getItem());
+
+		if (!EMCHelper.doesItemHaveEmc(item)) {
+			ctx.getSource().sendFailure(PELang.COMMAND_KNOWLEDGE_INVALID.translateColored(ChatFormatting.RED, item.getDisplayName()));
+			return 0;
+		}
+
+		switch (action) {
+			case LEARN -> {
+				if (provider.hasKnowledge(item)) {
+					ctx.getSource().sendFailure(PELang.COMMAND_KNOWLEDGE_LEARN_FAIL.translateColored(ChatFormatting.RED, player.getDisplayName(), item.getDisplayName()));
+					return 0;
+				} else {
+					provider.addKnowledge(item);
+					provider.sync(player);
+					ctx.getSource().sendSuccess(PELang.COMMAND_KNOWLEDGE_LEARN_SUCCESS.translateColored(ChatFormatting.GREEN, player.getDisplayName(), item.getDisplayName()), true);
+				}
+			}
+			case UNLEARN -> {
+				if (!provider.hasKnowledge(item)) {
+					ctx.getSource().sendFailure(PELang.COMMAND_KNOWLEDGE_UNLEARN_FAIL.translateColored(ChatFormatting.RED, player.getDisplayName(), item.getDisplayName()));
+					return 0;
+				} else {
+					provider.removeKnowledge(item);
+					provider.sync(player);
+					ctx.getSource().sendSuccess(PELang.COMMAND_KNOWLEDGE_UNLEARN_SUCCESS.translateColored(ChatFormatting.GREEN, player.getDisplayName(), item.getDisplayName()), true);
+				}
+			}
+			case TEST -> {
+				if (provider.hasKnowledge(item)) {
+					ctx.getSource().sendSuccess(PELang.COMMAND_KNOWLEDGE_TEST_SUCCESS.translateColored(ChatFormatting.GREEN, player.getDisplayName(), item.getDisplayName()), true);
+					return 1;
+				} else {
```

Comment:
Unneeded else, can just be implicit due to the fact the if statement always returns.